trestle/core/commands/merge.py

```diff
@@ -14,119 +14,131 @@
+            destination_model_filename.absolute(),
+            aliases_not_to_be_stripped=[target_model_alias])
+        """2. Load Target model. Target model could be stripped"""
+        try:
+            target_model_type = utils.get_target_model(element_path_list, merged_model_type)
+        except Exception as e:
+            raise err.TrestleError(
+                f'Target model not found. Possibly merge of the elements not allowed at this point. {str(e)}'
+            )
+        # target_model filename - depends whether destination model is decomposed or not
+        if (Path(os.getcwd()) / destination_model_alias).exists():
+            target_model_path = f'{os.getcwd()}/{destination_model_alias}/{target_model_alias}'
+        else:
+            target_model_path = target_model_alias
+
+        # if target model is a file then handle file. If file doesn't exist, handle the directory,
+        # but in this case it's a list or a dict collection type
+        if (Path(f'{target_model_path}.json')).exists():
+            target_model_filename = Path(f'{target_model_path}.json')
+            _, _, target_model_object = load_distributed.load_distributed(target_model_filename)
         else:
-            # List merge option for collection at the base level
-            destination_dir = cwd.parent if len(initial_path.parts) < len(cwd.parts) else cwd
-            destination = destination_dir / current_filename
-            visited_element = f'{current_alias}{path_sep}{const.ELEMENT_WILDCARD}'
-            if visited_element not in visited_elements:
-                visited_elements.add(visited_element)
-                self.out(f"{visited_element} (merges all files/subdirectories under {cwd} into \'{destination}\')")
-
-            # Go through each subdirectory in the collection and look for nested merge options
-            singular_alias = fs.get_singular_alias(current_alias, False)
-            singular_model = utils.get_inner_type(current_model)
-            for filename in sorted(cwd.glob(f'*{const.IDX_SEP}{singular_alias}')):
-                if Path.is_dir(filename):
-                    self._list_options_for_merge(
-                        filename,  # f'{current_alias}{path_sep}*',
-                        f'{current_alias}{path_sep}{singular_alias}',
-                        singular_model,
-                        f'{singular_alias}.json',
-                        initial_path=initial_path,
-                        visited_elements=visited_elements
-                    )
-
-    def _print_merge_option(self, visited_elements, element, source_path, destination_path) -> None:
-        if element not in visited_elements:
-            visited_elements.add(element)
-            self.out(f"{element} (merges all files/subdirectories under {source_path} into \'{destination_path}\')")
+            target_model_filename = Path(target_model_path)
+            collection_type = target_model_type.__origin__
```

Comment:
Avoid using internals here (typing_extensions.get_origin()

Reply:
`get_origin()` Seems to be not defined 😕

Reply:
Created #278 to cover off this issue as there are some interdependencies which are messy.